### PR TITLE
Finalise Rustup Windows Arm64 support

### PR DESCRIPTION
**Pull request by Alovchin91**
This PR finalises Windows Arm64 support by doing the following:

* ~~Removes the workaround to exclude Ring on Windows Arm64. Ring 0.17.x fully supports it.~~ Split-off into #3845
* ~Adds `aarch64-pc-windows-msvc` download link to cloud-invalidation.txt so that [this script](https://github.com/rust-lang/rust-forge/blob/master/blacksmith/src/lib.rs) could pick it up.~
  * ~Does the same for `aarch64-apple-darwin` because why not?~
* ~Updates the [Other Installation Methods page](https://forge.rust-lang.org/infra/other-installation-methods.html) to include Windows Arm64 version of Rustup.~ Split-off into #3854 
* Updates the [rustup.rs](https://rustup.rs) website to offer the arm64 version of Rustup on Windows Arm64.

The last part is the most "fragile" in my view because I don't know how the artefacts get published on rustup.rs, and detecting the arm64 CPU also required some async code that won't work in Firefox (but VS Code website’s arm64 detection doesn't work in Firefox either).

### CODE REVIEW

File: www/index.html
Comment (rami3l) [inline www/index.html:210]: Before we merge this, I think we should let @rust-lang/infra have a look at the artifact deployment: it looks like the URL is invalid now, but I guess the URL itself looks perfectly fine.

File: doc/user-guide/src/installation/other.md
Comment (rami3l) [inline doc/user-guide/src/installation/other.md]: I think this is not the intended way to do this. If Windows on ARM64 lacks proper host triple detection, we should fix it instead of documenting the wrong behavior.

Comment (Alovchin91) [reply doc/user-guide/src/installation/other.md]: No, it detects the host triple correctly. This was useful for x64 emulated Rustup, but is likely not needed anymore if the right artefact is published.

Comment (Alovchin91) [reply doc/user-guide/src/installation/other.md]: I’ll rephrase this section; right now it recommends to use x64 Rustup in all cases, but I think since Ring workaround is not needed anymore there’s no reason not to recommend the aarch64 Rustup to Windows Arm users.

File: www/index.html
Comment (Alovchin91) [reply www/index.html:210]: I think I’ll split off the removal of the Ring workaround into a separate PR then, so that it could be merged sooner. Then this PR will be exclusively about the delivery.

Comment (rami3l) [reply www/index.html:210]: @Alovchin91 Do you think it's okay to remove the trailing ` # skip-pr` hint in the following line:
 
https://github.com/rust-lang/rustup/blob/2a5a69e0914ff419554d684ca71eb1d72c72bcb3/ci/actions-templates/windows-builds-template.yaml#L23

... so that we might gain more confidence about the quality of the build? Currently no tests will be run for this artifact, but the fact that it compiles means something too at least...

File: doc/user-guide/src/installation/other.md
Comment (rami3l) [reply doc/user-guide/src/installation/other.md]: > No, it detects the host triple correctly. This was useful for x64 emulated Rustup, but is likely not needed anymore if the right artefact is published.

@Alovchin91 Let's track this in https://github.com/rust-lang/rustup/issues/3514.

File: www/index.html
Comment (Alovchin91) [reply www/index.html:210]: Yup, sounds good to me. Done!

File: doc/user-guide/src/installation/other.md
Comment (Alovchin91) [reply doc/user-guide/src/installation/other.md]: Rephrased. I'm actually not sure if #3514 is relevant here, since I don't understand the scenario where one would choose to use emulated Rustup with a native toolchain 🤔 It might be useful though to let Rustup upgrade itself from an emulated to a native version.

File: www/index.html
Comment (Alovchin91) [reply www/index.html:210]: Also split the Ring part off into #3845

**Comment by djc** [comment]
(This largely looks good to me, modulo one nit: would be nice to rebase away the merge commit.)

**Comment by Alovchin91** [comment]
> (would be nice to rebase away the merge commit.)

Done! I've been rebasing for years now so I didn't realise "Update fork" button would actually create a merge commit 😅

**Comment by rami3l** [comment]
> > (would be nice to rebase away the merge commit.)
> 
> Done! I've been rebasing for years now so I didn't realise "Update fork" button would actually create a merge commit 😅

@Alovchin91 You should've clicked "update with rebase". GitHub's default behavior is merging, unfortunately we cannot change it for now.

File: doc/user-guide/src/installation/other.md
Comment (rami3l) [inline doc/user-guide/src/installation/other.md]: One tiny nit:

```suggestion
* On Windows, download and run the `rustup-init.exe` built for the
  [`x86_64-pc-windows-msvc`] or [`aarch64-pc-windows-msvc`] target,
  depending on your OS architecture. In general, this is the build of
  `rustup` one should install on Windows. This will require the Visual C++
```

File: doc/user-guide/src/installation/other.md
Comment (rami3l) [reply doc/user-guide/src/installation/other.md]: @Alovchin91 Imagine one uses another app that installs Rustup. I don't know for Windows, but on macOS the guarantee is that once x64 is in the call chain, the rest of it is under emulation, so this will clearly cause problems. However this is out of the scope of this PR anyway, so don't worry 🙇

**Review by rami3l — APPROVED** [review]
LGTM! I'll ping t-infra again and let's see what we can do about https://win.rustup.rs/aarch64:
https://rust-lang.zulipchat.com/#narrow/stream/242791-t-infra/topic/.60win.2Erustup.2Ers.60.20should.20provide.20an.20.60aarch64.60.20installer

File: doc/user-guide/src/installation/other.md
Comment (Alovchin91) [reply doc/user-guide/src/installation/other.md]: Done!

**Comment by Alovchin91** [comment]
@rami3l Do you think it would make sense to split everything except rustup.rs website part off to a new PR that could be merged without waiting for infra? 🤔 It might already be helpful to have proper links in Other Installation Methods + the PR pipeline.

**Comment by rami3l** [comment]
> @rami3l Do you think it would make sense to split everything except rustup.rs website part off to a new PR that could be merged without waiting for infra? 🤔 It might already be helpful to have proper links in Other Installation Methods + the PR pipeline.

@Alovchin91 Yes, definitely. You can also use https://static.rust-lang.org/rustup/dist/aarch64-pc-windows-msvc/rustup-init.exe temporarily instead of https://win.rustup.rs/aarch64 first, then make a small PR to change that into the final URL. It's your call.

**Comment by Alovchin91** [comment]
@rami3l I think I'd prefer to keep it tidy and wait for the infra ☺️ Split off into #3854

**Comment by rami3l** [comment]
~~Status: Waiting for infra's deployment of https://win.rustup.rs/aarch64.~~ I just tested and the link seems to work without issues! Thanks a lot, @jdno!

A note to @Alovchin91: for the actual changes being made here to be in effect, we might have to wait until the next official release of Rustup. The latest version of the docs, however, can be found [here](https://rust-lang.github.io/rustup/devel/installation/other.html). Thank you for your patience :)

**Pull request merged**
